tools/doc/generate.js

```diff
@@ -80,14 +79,40 @@ fs.readFile(filename, 'utf8', async (er, input) => {
     .use(remark2rehype, { allowDangerousHTML: true })
     .use(raw)
     .use(htmlStringify)
-    .processSync(input);
-
-  const basename = path.basename(filename, '.md');
+    .process(input);
 
   const myHtml = await html.toHTML({ input, content, filename, nodeVersion });
+  const basename = path.basename(filename, '.md');
   const htmlTarget = path.join(outputDir, `${basename}.html`);
-  fs.writeFileSync(htmlTarget, myHtml);
-
   const jsonTarget = path.join(outputDir, `${basename}.json`);
-  fs.writeFileSync(jsonTarget, JSON.stringify(content.json, null, 2));
-});
+
+  return Promise.allSettled([
+    fs.writeFile(htmlTarget, myHtml),
+    fs.writeFile(jsonTarget, JSON.stringify(content.json, null, 2)),
+  ]);
+}
+
+main()
+  .then((tasks) => {
+    // Filter rejected tasks
+    const errors = tasks.filter(({ status }) => status === 'rejected')
+      .map(({ reason }) => reason);
+
+    // Log errors
+    for (const error of errors) {
+      console.error(error);
+    }
+
+    // Exit process with code 1 if some errors
+    if (errors.length > 0) {
+      return process.exit(1);
```

Comment:
technically you don't need a return here :P

Reply:
I know, but I think it's more clear and concistent whith early return pattern (I don't know the real name)